test/metabase/api/setup_test.clj

```diff
@@ -272,6 +274,39 @@
         (with-setup {:database {:engine "h2", :name db-name}}
           (is (db/exists? Database :name db-name)))))))
 
+(deftest create-superuser-only-once-test
+  (testing "POST /api/setup"
+    (testing "Check that we cannot create a new superuser via setup-token when a user exists"
+      (let [token (setup/create-token!)
+            body  {:token token
+                   :prefs {:site_locale "es_MX"
+                           :site_name   (mt/random-name)}
+                   :user  {:first_name (mt/random-name)
+                           :last_name  (mt/random-name)
+                           :email      (mt/random-email)
+                           :password   "p@ssword1"}}]
+
+        (is (setup/has-user-setup))
+
+        (with-redefs [setup/has-user-setup (let [value (atom false)]
+                                             (fn
+                                               ([] @value)
+                                               ([x] (reset! value x))))]
+
+          (is (not (setup/has-user-setup)))
+
+          (mt/discard-setting-changes [site-name
+                                       site-locale ;; anon-tracking-enabled
+                                       admin-email]
+            (fn []
```

Comment:
`mt/discard-setting-changes` is a macro that thunks its body so i don't think you want the manual `(fn [] ...)` bits here

Reply:
How embarrassing! 😆